src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -294,6 +294,13 @@ private static Rune ChangeCaseCultureAware(Rune rune, CultureInfo culture, bool
 
         public int CompareTo(Rune other) => this.Value - other.Value; // values don't span entire 32-bit domain; won't integer overflow
 
+        internal ReadOnlySpan<char> AsSpan(Span<char> buffer)
+        {
+            Debug.Assert(buffer.Length >= MaxUtf16CharsPerRune);
+            int charsWritten = EncodeToUtf16(buffer);
+            return buffer[..charsWritten];
```

Comment:
either use `buffer.Slice(0, charsWritten)` instead of `buffer[..charsWritten]` or define this method under `#if SYSTEM_PRIVATE_CORELIB`. I prefer `Slice` to allow using this method outside corelib.